### PR TITLE
Rails 5.2 tests fixes 1

### PR DESCRIPTION
**Pull request by yuenmichelle1**
Fixes controller tests a-m 

Changes: 
~~https://www.bigbinary.com/blog/rails-adds-support-for-recyclable-cache-keys~~
https://jetrockets.com/blog/rails-5-2-changes-in-callbacks (EDIT attached wrong article before. jetrockets blog is the correct one for these changes)

- In Rails 5.2 `changes` returns `{}` to get the equivalent we use `saved_changes`. 

- ~~Also updated dump_mailer spec to use `create(:project)` factory method in order to create project and so that `project.communication_emails` get set.~~ (EDIT: stubbed project resource and `communication_emails` return instead)

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: app/models/concerns/versioning.rb [hunk 1]
```diff
@@ -32,8 +32,7 @@ def save_version
     unless self.class.versioned_association.present?
       raise ConfigurationError, "Must call `versioned` DSL method if you include the Versioned module"
     end
-
-    if (changes.keys & self.class.versioned_attributes).present?
+    if (saved_changes.keys & self.class.versioned_attributes).present?
```
[the remark this answers is not in the document]
Comment (camallen) [reply app/models/concerns/versioning.rb:35]:
```suggestion

    if (saved_changes.keys & self.class.versioned_attributes).present?
```

File: spec/workers/concerns/dump_mailer_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 describe DumpMailer do
   let(:users) { create_list(:user, 2) }
   let(:owner) { users.sample }
-  let(:resource) { Project.new(id: 1, owner: owner) }
+  let(:resource) { create(:project, owner: owner) }
```
Comment (camallen) [inline spec/workers/concerns/dump_mailer_spec.rb:6]: This shouldn't need to be forced through a DB save to ensure the dump mailer specs work.

Ideally we don't pay the costs of creating factories and the associated records as they slow down the database. This is one of the reason the test suite is so slow on this code base. 

little changes like this add up over time so it's best to avoid this type of change.

File: app/models/concerns/versioning.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuenmichelle1) [reply app/models/concerns/versioning.rb:35]: @camallen I don't see the difference between the suggested change and the current version

File: spec/workers/concerns/dump_mailer_spec.rb (hunk 2 above)
Comment (yuenmichelle1) [reply spec/workers/concerns/dump_mailer_spec.rb:6]: I think that's fair.  Let me fix this test fix. I think we can get away with stubbing the resource and then have `communication_emails` return what we expect. 

For context, the two tests failing for dump mailer were the tests expecting to send an email to owner, which is set from the project's `communication_emails`.

File: app/models/concerns/versioning.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (camallen) [reply app/models/concerns/versioning.rb:35]: the suggested change preserves the original newline break
